### DISCUSSION TITLE
Need assistance uploading CSV via HTTP Listener POST method

### DISCUSSION THREAD

**Discussion by rjcdoc** [nextgenhealthcare/connect#5038] [Q&A]
I am unable to upload a CSV file via the HTTP Listener using a POST method call and save it to disk with File Writer.  The resulting CSV file I get is the whole POST Request and not just the content of the uploaded CSV file inside the content of the request body.  I am attaching an export of my channel and example File Writer result.

Channel Export - [api.xml.txt](https://github.com/nextgenhealthcare/connect/files/8102277/api.xml.txt)
File Writer CSV File output - [test.csv.txt](https://github.com/nextgenhealthcare/connect/files/8102278/test.csv.txt)

**Comment by pacmano1** [comment]
If you are just posting a text file to the endpoint set the Message Content to "Plain Body" on your listener, not "XML Body".

**Reply by rjcdoc** [reply]
I changed to "plain Body" and it still puts the full POST request data in the file, not the data from the file.  see below:
```
----------------------------866539694895748143044617

Content-Disposition: form-data; name="file"; filename="intake_data_test.csv"

Content-Type: text/csv



patient_id,patient_first_name,patient_middle_name,patient_last_name,patient_dob,patient_sex,patient_phone,patient_address,patient_city,patient_state,patient_zip,icd_codes,provider_name,provider_group_name,provider_direct_address
00001,first,middle,last,2000-01-01,M,123-456-7890,123 any street, city,state,12345,code,name,group,email@address.com
00002,first,middle,last,2000-01-01,M,123-456-7890,123 any street, city,state,12345,code,name,group,email@address.com
00003,first,middle,last,2000-01-01,M,123-456-7890,123 any street, city,state,12345,code,name,group,email@address.com
00004,first,middle,last,2000-01-01,M,123-456-7890,123 any street, city,state,12345,code,name,group,email@address.com
00005,first,middle,last,2000-01-01,M,123-456-7890,123 any street, city,state,12345,code,name,group,email@address.com

----------------------------866539694895748143044617--
```

**Reply by pacmano1** [reply]
It matters how the sender is posting  to your channel.   Don't have them sent multi-part.    If you must accept multi-part, then you need to just pull the files out of the XML blob you receive a transformer.
